client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/db/constraint/H2ConstraintExtractor.java

```diff
@@ -0,0 +1,132 @@
+
+        List<DbTableConstraint> tableCheckExpressions = new LinkedList<>();
+
+        String tableSchema = schemaDto.name;
+        for (TableDto tableDto : schemaDto.tables) {
+            String tableName = tableDto.name;
+            Statement statement = connectionToH2.createStatement();
+            ResultSet constraints = statement.executeQuery("Select * From INFORMATION_SCHEMA.CONSTRAINTS "
+                    + " where CONSTRAINTS.TABLE_SCHEMA='" + tableSchema + "' and CONSTRAINTS.TABLE_NAME='" + tableName + "'");
+
+            while (constraints.next()) {
+                String constraintType = constraints.getString("CONSTRAINT_TYPE");
+                String sqlCheckExpression = constraints.getString("CHECK_EXPRESSION");
+                String columnList = constraints.getString("COLUMN_LIST");
+
+                if (constraintType.equals("UNIQUE")) {
+                    assert (sqlCheckExpression == null);
+                    List<String> uniqueColumnNames = Arrays.stream(columnList.split(",")).map(String::trim).collect(Collectors.toList());
+                    DbTableUniqueConstraint uniqueConstraint = new DbTableUniqueConstraint(tableName, uniqueColumnNames);
+                    tableCheckExpressions.add(uniqueConstraint);
+
+                } else if (constraintType.equals("REFERENTIAL")) {
+                    /**
+                     * This type of constraint is already handled by
+                     * JDBC Metadata
+                     **/
+                    continue;
+                } else if (constraintType.equals("PRIMARY KEY") || constraintType.equals("PRIMARY_KEY")) {
+                    /**
+                     * This type of constraint is already handled by
+                     * JDBC Metadata
+                     **/
+                    continue;
+                } else if (constraintType.equals("CHECK")) {
+                    assert (columnList == null);
+
+                    DbTableCheckExpression constraint = new DbTableCheckExpression(tableName, sqlCheckExpression);
+                    tableCheckExpressions.add(constraint);
+
+                } else {
+                    throw new RuntimeException("Unknown constraint type : " + constraintType);
+                }
+
+            }
+
+            statement.close();
+
+        }
+
+        return tableCheckExpressions;
+    }
+
+
+    /**
+     * Appends all Column constraints (i.e. CHECK contraints) to the DTO
+     *
+     * @param connectionToH2 a connection to a H2 database
+     * @param schemaDto
+     * @throws SQLException if the connection to the database fails
+     */
+    private List<DbTableConstraint> extractColumnConstraints(Connection connectionToH2, DbSchemaDto schemaDto) throws SQLException {
+        String tableSchema = schemaDto.name;
+        List<DbTableConstraint> columnConstraints = new LinkedList<>();
+        for (TableDto tableDto : schemaDto.tables) {
```

Comment:
any specific reason for using `LinkedList` instead of `ArrayList`? I always thought that `LinkedList` should be avoided as more expensive / slower, but maybe I am wrong?

Reply:
Thanks. I replaced all references to LinkedList with ArrayList